Bottom: c5f857e9afbae35f699258fe8bb1aff83896e5e8
Top:    23712a6a6eeac910e5524c9d6fcf6827f653c8c3
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-04-27 22:36:39 +0800

Refresh of vhost_net-avoid-unnecessary

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 6b91a40..289054d 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -759,7 +759,7 @@ static void handle_rx(struct vhost_net *net)
 			 * TODO: support TSO.
 			 */
 			if (copy_to_iter(&hdr, sizeof(hdr),
-					 &fixup) != sizeof(hdr)) {
+					 &msg.msg_iter) != sizeof(hdr)) {
 				vq_err(vq, "Unable to write vnet_hdr "
 				       "at addr %p\n", vq->iov->iov_base);
 				goto out;
@@ -778,8 +778,8 @@ static void handle_rx(struct vhost_net *net)
 			continue;
 		}
 		if (mergeable) {
-			/* Header came from socket; we'll need to patch
-			 * ->num_buffers over if VIRTIO_NET_F_MRG_RXBUF
+			/* We'll need to patch ->num_buffers over if
+			 * VIRTIO_NET_F_MRG_RXBUF
 			 */
 			iov_iter_advance(&fixup, sizeof(hdr));
 		}
